### DISCUSSION TITLE
Creating a Notification with pyodide

### DISCUSSION THREAD

**Discussion by TobiasHT5** [pyodide/pyodide#4394] [Q&A] [answered]
Hey there. So I've been trying out pyodide in the REPL and love how it interacts with the broswer.
I was testing out how to work with Notifications and I did the most part right, ie, getting the permissions and all, but my notification ain't showing up when I raise it. Am I missing something?

<img width="835" alt="Screenshot 2024-01-20 at 16 11 18" src="https://github.com/pyodide/pyodide/assets/35004147/7790feb6-176a-4475-b8d4-a47e353f0514">

**Comment by hoodmane** [comment, marked as answer]
Try:
```py
Notification.new("hello", body="Hello")
```

**Reply by TobiasHT5** [reply]
@hoodmane have done that but Notification isn't showing up.

**Reply by hoodmane** [reply]
It worked for me, can you make a complete reproduction?

**Reply by TobiasHT5** [reply]
@hoodmane apparently I was using chrome on my macbook and it couldn't show notifications. When I used Safari, the notifications appeared. So perhaps it's a platform thing.

**Reply by hoodmane** [reply]
Are you creating the notification inside of an event handler? I think there's a restriction that notifications be created in response to a user action.

**Reply by TobiasHT5** [reply]
@hoodmane I was just writing prompts in the web REPL. However I've failed to figure out how to add actions to the notification.
Was writing something like
```python
>>> Notification.new("Hello",body="Hello World", actions = [{"action":"like","title":"like"}])
```

**Reply by hoodmane** [reply]
Well if I try to pass an actions parameter I see:
```
Uncaught TypeError: Failed to construct 'Notification': Actions are only supported for persistent notifications shown using ServiceWorkerRegistration.showNotification().
```

**Reply by TobiasHT5** [reply]
@hoodmane Thanks for this. Guess I'll keep experimenting with stuff till I find something..
